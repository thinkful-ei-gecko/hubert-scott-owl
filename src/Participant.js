import React from 'react';


//props is expected to be an Object containing paticipant id, avatar link, name, boolean inSession, boolean, onStage
export default function Participant(props) {
    return (
        <li className="participant" key={props.id}>
            <img src={props.avatar} alt="placeholder" className="avatar"></img>
            <div className="participant-info">
                <h3 className="participant-name">{props.name}</h3>
                <div className={props.inSession ? 'in-session': 'left-session'}></div>
                <p>{props.onStage ? 'on stage' : ''}</p>
            </div>
        </li>
    );
}