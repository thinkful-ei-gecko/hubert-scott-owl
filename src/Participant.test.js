import React from 'react';
import ReactDOM from 'react-dom';
import Participant from './Participant';
import renderer from 'react-test-renderer'


describe('<Participant />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<Participant id = 'a' name = 'frank' avatar = 'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1' inSession={true} onStage={true} />, div);

        ReactDOM.unmountComponentAtNode(div);
    });

});

describe('<Participant />', () => {
    it('renders UI correctly', () => {
        const tree = renderer.create(<Participant id = 'a' name = 'frank' avatar = 'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1' inSession={true} onStage={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});