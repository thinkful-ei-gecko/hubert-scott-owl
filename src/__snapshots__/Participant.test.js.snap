// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Participant /> renders UI correctly 1`] = `
<li
  className="participant"
>
  <img
    alt="placeholder"
    src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
  />
  <h3>
    frank
  </h3>
  <div
    className="in-session"
  />
  <p>
    on stage
  </p>
</li>
`;
